docs/multicluster/architecture.md

```diff
@@ -0,0 +1,87 @@
+# Antrea Multi-cluster Architecture
+
+Antrea Multi-cluster controller focuses on service discovery of containers across
```

Comment:
Maybe remove "of containers". It is k8s cluster, no other workloads.

Reply:
Should we say: "Antrea multi-cluster implements Multi-cluster Service API"? It is not just service discovery right?

And add a link to Multi-cluster Service API somewhere.

Reply:
done